### ISSUE TITLE
[v4][no-await-sync-query] False positives on appearance/disappearance examples

### ISSUE THREAD

**Issue by epmatsw** [testing-library/eslint-plugin-testing-library#304]
In https://testing-library.com/docs/guide-disappearance, there's some techniques outlined for waiting for elements to be removed:

```js
test('movie title appears', async () => {
  // element is initially not present...

  // wait for appearance inside an assertion
  await waitFor(() => {
    expect(getByText('the lion king')).toBeInTheDocument()
  })
})
```

```
test('movie title no longer present in DOM', async () => {
  // element is removed
  await waitForElementToBeRemoved(() => queryByText('the mummy'))
})
```

Both of these examples are flagged by `no-await-sync-query`, and they probably shouldn't be.

**Comment by Belco90** [comment]
Hey @epmatsw, thank you very much for your feedback on v4! I think I got a similar issue to this one with `no-await-sync-query`, and I believe I know where the issue is located in the code.
